FormalConjectures/Paper/CasasAlvero.lean

```diff
@@ -0,0 +1,95 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Casas-Alvero Conjecture
+
+*Reference:*
+* [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/math/0605090)
+* [MathOverflow](https://mathoverflow.net/questions/27851)
+
+The Casas-Alvero conjecture states that if a univariate polynomial P of degree d over a field
+of characteristic zero shares a root with each of its derivatives up to order d-1, then P must
+be of the form (X - α)^d for some α.
+
+The conjecture has been proven for:
+* Degrees d ≤ 8
+* Degrees of the form p^k where p is prime
+* Degrees of the form 2p^k where p is prime
+
+The conjecture is false in positive characteristic p for polynomials of degree d > p.
+
+The conjecture is now claimed to be proven in this paper:
+* [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/2505.18000)
+
+-/
+
+open Polynomial
+
+/--
+A polynomial P satisfies the Casas-Alvero property if it shares a root with each
+of its derivatives up to order d-1, where d is the degree of P.
+-/
+def HasCasasAlveroProp {F : Type} [Field F] (P : F[X]) : Prop :=
+  ∀ i ∈ Finset.range P.natDegree, ∃ α : F, IsRoot P α ∧ IsRoot (derivative^[i] P) α
```

Comment:
Maybe switch F to K here too, and extract the `variable`s `{K : Type*} [Field K]` (I'm not sure whether there's a convention in this repo to use `Type*` or `Type`; `Type` may be closer to ZFC which is the generally accepted foundation? But I expect most questions' answer (true or false) are independent of which universe the objects are in.) Might be a good idea to extract `[CharZero K] (P : K[X]) (hP : Monic P)` too (you'll need `include hP`).

It might be better to add a version using `¬ IsCoprime P (derivative^[i] P)`, and show the conjecture is equivalent using both definitions (as an API lemma).